### PR TITLE
chore(styles): introduced RTL dist package

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)

#3984

### Description

In order to properly support RTL adopters, this change uses `rtlcss` to create an RTL equivalent of the styles package that can be easily imported into adopters' sites.

**Use:**

`import '@carbon/ibmdotcom-styles/dist/ibm-dotcom-styles-rtl.css';`

### Changelog

**New**

- Introduced an RTL styles package

### CODE REVIEW

**Review by asudoh — COMMENTED** [review]
Thank you for jumping in @IgnacioBecerra! I think we need a separate task for the RTL version, because `cleanCSS()` strips out RTLCSS directives and thus we need to apply RTLCSS before `cleanCSS()` runs.

**Comment by jeffchew** [comment]
> Thank you for jumping in @IgnacioBecerra! I think we need a separate task for the RTL version, because `cleanCSS()` strips out RTLCSS directives and thus we need to apply RTLCSS before `cleanCSS()` runs.

Agreed, would be cleaner to keep the two separate anyway.

**Review by asudoh — COMMENTED** [review]
Thank you for splitting the Sass tasks! Just a nit (including naming), I'd create `sass:ltr` and `sass:rtl` tasks and make `sass` task run `sass:ltr` and `sass:rtl` in parallel. You can see what I mean by `packages/web-components/gulpfiles.js` file and `packages/web-components/gulp-tasks` directory.

File: packages/styles/gulp/tasks/rtlSass.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
```
Comment (asudoh) [inline packages/styles/gulp/tasks/rtlSass.js:1]: Is this file still needed...?

**Review by asudoh — COMMENTED** [review]
Thank you for the update @IgnacioBecerra!

File: packages/styles/gulp/build/default.js [hunk 2]
```diff
@@ -8,10 +8,17 @@
...
 const gulp = require('gulp');
+const sass = require('../tasks/sass');
 
 /**
  * Gulp task export
  *
  * @module default
  */
-module.exports = gulp.task('default', gulp.series('clean', 'sass'));
+module.exports =
```
Comment (asudoh) [inline packages/styles/gulp/build/default.js:18]: What did you want to point `module.exports` to...?

**Review by asudoh — COMMENTED** [review]
Thank you for another update @IgnacioBecerra!

File: packages/styles/gulp/build/default.js (hunk 2 above)
Comment (jeffchew) [reply packages/styles/gulp/build/default.js:18]: @IgnacioBecerra can you take a look? The module.exports looks off like what @asudoh mentioned.

Comment (IgnacioBecerra) [reply packages/styles/gulp/build/default.js]: @asudoh @jeffchew Separate tasks into their own files and exported the default tasks correctly this time around. 👍

File: packages/styles/gulp/tasks/ltr-sass.js [hunk 3]
```diff
@@ -49,6 +62,6 @@ function _sass() {
 /**
  * Gulp task export
  *
- * @module sass
+ * @module sass-ltr
  */
-module.exports = gulp.task('sass', _sass);
+module.exports = gulp.task('sass-ltr', ltr);
```
Comment (jeffchew) [inline packages/styles/gulp/tasks/ltr-sass.js:67]: @IgnacioBecerra super nitpick, can you name the files to match the task name? `sass-ltr.js` and `sass-rtl.js`.

Comment (IgnacioBecerra) [reply packages/styles/gulp/tasks/ltr-sass.js]: Done!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @IgnacioBecerra for all the updates!
